Correctly unload plugins

ZarTek-Creole opened the pull request:
Plugins unloading is now done before deleting the parent namespace.
There were bugs because of this.

Creation of '**deinit_plugin** <plugin_name>' allows to unload a plugin.

Creation of '**deinit_allplugins**' allows to unload all plugins.

The '**deinit_allplugins**' is run in '**deinit_ng**' before unloading pzs.

in '**init_plugins**' when a module fails to load 'deinit_plugin' is run to clean the failed plugin.

Unloading a plugin by '**deinit_plugin**' does this:
- Execute '**deinit**' of the plugin if present
- Removes all types of binds existing in the plugins
- Removes all '**timers**' existing in the plugins
- Removes all '**utimers**' existing in the plugins
- Remove a package declaration if the plugin provides a package
- Remove the plugin namespace

deinit_plugin accepts two types of value:
- Namespace: **deinit_plugin** ::ngBot::plugin::Plugin_Name
- Plugin name: **deinit_plugin** Plugin_Name

To sum up: all plugins are correctly unloaded even if their deinit is badly defined.
the deinit no longer needs to 'kill' its (u) timers, unbind the binds, etc.
deinit of the plugin is now useful for performing some tasks before closing. Closing of its flows, saving in files, saving ...


Example of a rehash:
```

[22:07:40] #MalaGaM# rehash
Rehashing.
[22:07:40] Writing user file...
[22:07:39] Writing channel file...
[22:07:39] Rehashing ...
[22:07:39] [ngBot] info :: UnBind pub: 'Trigger' from MalaGaM-AutoWipeNuked
[22:07:39] [ngBot] info :: UnBind time: 'time:proc' from MalaGaM-AutoWipeNuked
[22:07:39] [ngBot] info :: UnBind pub: 'Trigger' from MalaGaM-Quotat
[22:07:39] [ngBot] info :: UnBind time: 'time:proc' from MalaGaM-Quotat
[22:07:39] [ngBot] info :: UnBind msg: 'Trigger' from TVMaze
[22:07:39] [ngBot] info :: UnBind pub: 'Trigger' from TVMaze
[22:07:39] [ngBot] info :: UnBind msg: 'SearchDcc' from psxc-IMDb
[22:07:39] [ngBot] info :: UnBind pub: 'Search' from psxc-IMDb
[22:07:39] Writing channel file...
[22:07:39] Listening for telnet connections on 0.0.0.0 port 3333 (all).
[22:07:39] bseen1.4.2c:  -- Bass's SEEN loaded --
[[22:07:39] Writing channel file...
[22:07:39] Userfile loaded, unpacking...
[22:07:39] [ngBot] Detected glftpd 2, running in standard mode.
[22:07:39] [ngBot] Number of logs found: 4
[22:07:39] [ngBot] Loading theme "scripts/pzs-ng/themes/MalaGaM.zst".
[22:19:34] [ngBot] [SUCCESS->MalaGaM-IncLink] Loaded successfully (Version: 0.1 - 2021-04-21).
[22:19:34] [ngBot] [SUCCESS->MalaGaM-MediaInfo] Loaded successfully (Version: 0.1 - 2021-04-21).
[22:19:34] [ngBot] [SUCCESS->MalaGaM-AutoWipeNuked] Loaded successfully (Version: 0.1 - 2021-04-21).
[22:19:34] [ngBot] [SUCCESS->MalaGaM-Tools] Loaded successfully (Version: 0.1 - 2021-04-21).
[22:19:34] [ngBot] [SUCCESS->MalaGaM-PreTime] Loaded successfully (Version: 0.1 - 2021-04-24).
[22:19:34] [ngBot] [SUCCESS->MalaGaM-Quotat] Loaded successfully (Version: 0.1 - 2021-04-21).
[22:19:34] [ngBot] [SUCCESS->MalaGaM-MySQL] Loaded successfully (Version: 0.1 - 2021-04-21).
[22:19:34] [ngBot] TVMaze :: Loaded successfully (Version: 20200223).
[22:19:34] [ngBot] psxc-imdb :: Loaded successfully. (Modes Enabled: Logging, Pre, Find)
[22:19:34] [ngBot] Plugins Loaded: MalaGaM-IncLink, MalaGaM-MediaInfo, MalaGaM-AutoWipeNuked, MalaGaM-Tools, MalaGaM-PreTime, MalaGaM-Quotat, MalaGaM-MySQL, TVMaze, psxc-IMDb.
[22:19:34] [ngBot] Loaded successfully!

```

ZarTek-Creole committed:
Correctly unload plugins

hrxcodes commented on sitebot/ngBot.tcl:
```diff
@@ -242,7 +242,7 @@ namespace eval ::ngBot {
 	}
 
-	proc init_plugins {} {
+		proc init_plugins {} {
```
This should be left untouched, already has correct indentation

hrxcodes commented on sitebot/ngBot.tcl:
```diff
@@ -252,13 +252,13 @@ namespace eval ::ngBot {
 				0 { lappend loaded [namespace tail $plugin] }
 				default {
-					putlog "\[ngBot\] [namespace tail $plugin] Error :: Unable to load plugin: $error"
+					putlog "\[ngBot\] [namespace tail $plugin] Error :: Unable to load plugin: $error\n$::errorInfo"
```
Why removing the errorInfo?

ZarTek-Creole replied:
where is  removed? i have added \n$::errorInfo for more information on error

hrxcodes replied:
Ah sorry, my mixup

ZarTek-Creole committed:
Fix indentation for @hrxcodes comment

This should be left untouched, already has correct indentation

_Originally posted by @hrxcodes in https://github.com/pzs-ng/pzs-ng/pull/67#r619712312_
